replica/out_of_space_controller.hh

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#pragma once
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/sharded.hh>
+#include <seastar/util/bool_class.hh>
+
+#include "utils/disk_space_monitor.hh"
+#include "utils/updateable_value.hh"
+
+namespace replica {
+
+// Simple controller to notify database it is in the critical disk utilization zone. The action
```

Comment:
nit: it's not only the database that is notified, this could be generalize to something like `notify subscribers, such as the database`